Add warning alert for archived, locked, and commentID posts

jamesmaa opened the pull request:
I'm actually having trouble messing around with themeify css. Could someone teach me? 
Otherwise:
**locked**
![image](https://cloud.githubusercontent.com/assets/2382537/17959053/bd891b76-6a52-11e6-8512-e5acf852e66c.png)
**single comment**
![image](https://cloud.githubusercontent.com/assets/2382537/17959057/c295d4ec-6a52-11e6-9373-93db5fcc22d0.png)
**archived**
![image](https://cloud.githubusercontent.com/assets/2382537/17959059/c6723510-6a52-11e6-8404-957137675000.png)
**night mode**
![image](https://cloud.githubusercontent.com/assets/2382537/17959061/caeb384e-6a52-11e6-9e67-0abd8cd6befd.png)

👓  @uzi @phil303

uzi commented on src/app/components/CommentsPage/CommentsPageTools/index.jsx:
```diff
@@ -8,8 +8,44 @@ import CommentReplyForm from 'app/components/Comment/CommentReplyForm';
+  const postLoaded = !!post;
+  if (!postLoaded) {
+    return null;
+  }
```
🔕 Could also just go with `if (!post)` since postLoaded isn't used otherwise... but it's ok if you think it helps readability.

uzi commented on src/app/components/CommentsPage/CommentsPageTools/index.jsx:
```diff
@@ -8,8 +8,44 @@ import CommentReplyForm from 'app/components/Comment/CommentReplyForm';
+    message = <p> Comments are locked </p>;
+  } else if (subredditDetail && subredditDetail.user_is_banned) {
+    message = <p> You are banned from commenting in this community for now </p>;
+  }
```
Hmm... some messages are in <p> and some are not... is there a reason for the disparity?

jamesmaa replied:
oops late night blunder

uzi commented on src/app/components/CommentsPage/CommentsPageTools/index.jsx:
```diff
@@ -8,8 +8,44 @@ import CommentReplyForm from 'app/components/Comment/CommentReplyForm';
 
-export default ({ replying, currentPage, id, onSortChange }) => {
+  let message;
+  const subredditDetail = post.subredditDetail;
```
Could do `const { subredditDetail } = post;` instead.

uzi commented:
💅 Nice!  Three birds, meet one stone!  Just a couple of nits, but this looks good!

jamesmaa commented:
💇 Thanks for reviewing!

phil303 commented on src/app/components/CommentsPage/CommentsPageTools/index.jsx:
```diff
@@ -8,8 +8,43 @@ import CommentReplyForm from 'app/components/Comment/CommentReplyForm';
 import { SORTS } from 'app/sortValues';
 
+const renderThreadNotice = (post, hasSingleComment) => {
+  if (!post) {
```
Where possible, one thing that I've found that makes jsx a bit clearer is to move these types of checks into the parent render function. So for instance with this particular one, the code below (~line 84) would look something like:

```jsx
{ post ? renderThreadNotice(post, hasSingleComment) : null }
```

This has the nice effect of letting future readers avoid diving down into functions unnecessarily and making it very apparent what this element requires in order to be visible.

phil303 commented on src/app/components/CommentsPage/CommentsPageTools/styles.less:
```diff
@@ -2,7 +2,35 @@
 @import (reference) '~app/less/themes/themeify';
 
+.themed-alert(@base-color, @background-color, @text-color) {
+  .themeify(
```
You're pretty close here.  We actually have color constants for both day mode and night mode defined in separate files. Since yours look new and don't yet exist, you'll have to define them in both `app/less/themes/dayMode.less` and its counterpart `nightMode.less`. When that's done, your code will just "do the right thing" when in either mode if it looks something like this:

```less
.themeify({
  border-color: @theme-warning-border;
  background-color: @theme-warning-body-color;
  color: @theme-warning-body-text-color;
});
```

phil303 replied:
One thing to be careful of - we do have a color style guide [here](https://www.dropbox.com/s/f9pc60gj1bkmd3e/reddit_brandbook.pdf?dl=0) (page 17). Since some of these colors are not part of it, it might be good to get some quick input from Jonathan or Derek. They might have some good feedback on the color choices.

jamesmaa replied:
I copied the styles from 1X:
https://github.com/reddit/reddit-mobile/blob/master/assets/less/components/alerts.less

phil303 replied:
Ah interesting. If you don't mind, it'd still be good to do a quick check in with them. This might've been before the style guide was introduced and would be a good opportunity to make sure we're aligned.

phil303 commented on src/app/components/CommentsPage/CommentsPageTools/index.jsx:
```diff
@@ -8,8 +8,43 @@ import CommentReplyForm from 'app/components/Comment/CommentReplyForm';
+    );
+  }
+
+  return null;
```
nit, returning null isn't really necessary in this case

phil303 commented:
:nail_care:

jamesmaa commented:
💇 
Re styles: the styles are consistent with 1X. Is that good enough or should I ask design?

jamesmaa commented:
UPDATE: So I talked to design and they are okay with the color scheme. From a UX perspective though they want us to follow the UX of the mobile apps, mainly
1. When a post is archived, the warning box doesn't show. They don't think it's useful information.
2. They want the upvote/downvote/reply buttons to be removed if a post is archived or locked.

I don't know if that is within the scope of this commit. Thoughts?

jamesmaa commented:
UPDATES
- Moved all styles to `dayMode.less` and `nightMode.less`
- Disabled voting and removed downvote on archived posts
- Removed "Post is archived" message per design team's request

Screenshots:
**archived**
![image](https://cloud.githubusercontent.com/assets/2382537/17994244/ab4b9a2e-6b0b-11e6-8e7b-5cbe91a974b3.png)
![image](https://cloud.githubusercontent.com/assets/2382537/17994250/b0a1b4b8-6b0b-11e6-9b76-a1a22ae3d84e.png)

**locked**
![image](https://cloud.githubusercontent.com/assets/2382537/17994254/b90ed374-6b0b-11e6-8a82-81f0816fd6d3.png)
![image](https://cloud.githubusercontent.com/assets/2382537/17994256/bcf85316-6b0b-11e6-9d75-714a435c3a44.png)

**single comments**
![image](https://cloud.githubusercontent.com/assets/2382537/17994261/c622a144-6b0b-11e6-87d3-b23459c232e6.png)

💇

uzi commented on src/app/components/Comment/CommentTools/index.jsx:
```diff
@@ -85,12 +87,13 @@ const renderSeashells = tooltipId => (
     classPrefix='CommentTools'
     score={ score }
     scoreHidden={ scoreHidden }
+    hideDownvote={ hideDownvote }
```
Hmm... it's votingDisabled in commentTools, but hideDownvote here?  Why only hide the downvote?  Seems we'd want to hide both and only show the score, right?  That's what we did on 1X at least, iirc.

jamesmaa replied:
oh native-mobile, they show the upvote to indicate the score.
![image](https://cloud.githubusercontent.com/assets/2382537/17995025/e5f79bd0-6b12-11e6-9f2d-a118d3509d5e.png)

uzi commented on src/app/components/Comment/index.jsx:
```diff
@@ -236,6 +247,16 @@ const userSelector = state => state.user;
+  const postId = `t3_${state.platform.currentPage.urlParams.postId}`;
+  const post = state.posts[postId];
+  return post ? (post.archived) : true;
+};
```
You said "They want the upvote/downvote/reply buttons to be removed if a post is archived or locked" ... so then commentingDisabledSelector and votingDisabledSelector should be the same.  Should we consolidate?

jamesmaa replied:
A locked post should be able to upvote/downvote but not comment right? That's why I had two separate variables

uzi commented on src/app/components/Vote/index.jsx:
```diff
@@ -31,7 +31,7 @@ export function renderUpvote (props) {
     >
       <input type='hidden' value={ formDirection } name='direction' />
-      <button className={ cls } type='submit' />
+      <button className={ cls } type='submit' disabled={ hideDownvote } />
```
Yep, just confused about hiding only downvoting when I don't think they'll be able to vote at all.

jamesmaa replied:
the logic is that when you hide the downvote, the renderUpvote button should render a disabled upvote

uzi commented:
💅 Looking great, just want to double-check about the voting situation. Thanks!

jamesmaa commented:
💇 The voting situation is that through talking with design, they want the UX to align with the native mobile experience, so they still want an upvote next to the score

uzi commented:
:fish: I double checked on desktop and you're right about the locked post.  Thanks again for the help!
